README.md

```diff
@@ -1,26 +1,57 @@
 # signex
-Sign messages with private key
+Secure messages through digests and signing.
+
+SignEx can work with any message which has:
+- content that can be represented as a binary, such as an HTML body.
+- metadata that can be represented as a map, such as HTML headers.
+
+Signing a message consists of two steps
+
+- Generating a digest of the content
+- Signing the message metadata which includes the content digest.
+
 
 ## Usage
 
-### To sign
+### Verifying generic messages
 
 ```elixir
-private_key = # read private key
-public_key = # read public key
-{:ok, signature, signed_message} = SignEx.sign("my message", private_key, public_key)
+:ok = SignEx.verify_signature(metadata, signature_params, keystore)
+:ok = SignEx.verify_digest(body, digest)
+
+# Or combined
+# will assume that there is a "digest" key in the meta data that is used to confirm the request
+:ok = SignEx.verify(body, metadata, signature_params, keystore)
```

Comment:
we should also add an interface for verifying messages with no body (ie. GET requests):

```elixir
:ok = SignEx.verify(metadata, signature_params, &fetch_key_from_id)
```

Also we should add a definition for both methods that allows the `key_id` to be passed in as a string rather than a callback function.